### PR TITLE
convert html writer to html 5.

### PR DESCRIPTION
**Pull request by tsteven4**
this also fixes a bug where we could fail to entitize wpt->description.
this also fixes a bug where we could create illegal fragments and id
attributes.

There are some minor changes in rendering with firefox related to margins in p elements.

Every gpx reference file we have was converted to html, and the html was checked with tidy.  There are two files that have warnings, and no errors.
```
reference/tpo-sample3.gpx:
line 250 column 1 - Warning: <a> converting backslash in URI to slash
```
We are interpreting 'w:\photo.jpg' in tpo-sample3.tpo as a link.
```
reference/gc/GCGCA8.gpx:
line 45 column 154 - Warning: discarding unexpected </a>
line 45 column 158 - Warning: inserting implicit <p>
line 45 column 158 - Warning: trimming empty <p>
```
GCGCA8.gpx has two unpaired end tags (entitized) in a groundspeak:long_description element. Also see #895.

The html files produced by text.test were checked with https://validator.w3.org.  Related to GCGCA8.gpx we see these warnings/errors:
```
Warning: Consider adding a lang attribute to the html start tag to declare the language of this document.
Warning: The border attribute is obsolete. Consider specifying img { border: 0; } in CSS instead.
Error: Stray end tag a.
Error: No p element in scope but a p end tag seen.
```

### CODE REVIEW

**Comment by robertlipe** [comment]
Well done.

The html inside PQs is strange and doubly so for the age of our examples.
They predate html5 by far and the html is meant to be embedded inside a
geocache listing page, but not inside an iframe. That's why there's not a
top level html tag, for example. Further the html is often rendered by a
handheld gps that doesn't have a full html browser so there are all kinds
of unwritten rules.

In short, don't stress about the html inside our GC reference files. The
html target was originally written to take those files and render them
viewable on PDAs before the GPSes could.

**Comment by tsteven4** [comment]
QXmlStreamWriter is nice for XML, and QTextStream seems better than gbf*, but for HTML Qt doesn't seem to have a good solution.  QTextDocument in 6.3.2 supports a subset of HTML 4.  XHTML doesn't seem to have much share.
